pyomo/core/base/component.py

```diff
@@ -1157,15 +1163,236 @@ def __ne__(self, other):
+                        self._partial_cuid_from_slice_info(slice_info)
+                        )
+                parent = comp.parent_block()
+                for cid in self._generate_cuid(parent, context=context):
+                    # Generate _cid entries for parent (non-slice)
+                    # components. This is the only place `context` gets
+                    # used, as the slice does not access any "components"
+                    # in its call call stack.
+                    yield cid
+            elif call == IndexedComponent_slice.get_item:
+                # Need to parse index to get potential slice
+                index = arg
+                # Note that this assumes we will never have two get_item
+                # calls in a row.
+            elif call == IndexedComponent_slice.call:
+                if len(arg) != 1:
+                    raise NotImplementedError(
+                            "Cannot create a CUID from a slice with a "
+                            "call that has multiple arguments. Got "
+                            "arguments %s." % (arg,)
+                            )
+                # Cache argument of a call to `component`
+                name = arg[0]
+                if kwds != {}:
+                    raise NotImplementedError(
+                            "Cannot create a CUID from a slice with a "
+                            "call that contains keywords. Got keyword "
+                            "dict %s." % (kwds,)
+                            )
+            elif call == IndexedComponent_slice.get_attribute:
+                if name is not None:
+                    # This only happens if IndexedComponent_slice.call
+                    # was encountered.
+                    if arg != 'component':
+                        raise NotImplementedError(
+                            "Cannot create a CUID from a slice with a "
+                            "call to any method other than `component`. "
+                            "Got %s." % arg
+                            )
+                    else:
+                        # name is the attr we actually want to get
+                        arg = name
+                        # Reset name to None
+                        name = None
+                #if count == 1:
+                     # We have encountered a get_attr at the top
+                     # of our stack. This has been handled differently
+                     # depending on how the CUID was constructed:
+                     # - Constructing a CUID from an indexed component
+                     #   will treat its indices as wildcards.
+                     # - Constructing from a string will yield a valid
+                     #   CUID that will treat the indexed component as
+                     #   itself (rather than a slice over its data).
+                     # This is an issue with CUID that should be
+                     # addressed.
```

Comment:
This is an inconsistency with CUIDs. If `m.b.v` is an `IndexedVar`, 
```python
ComponentUID(m.b.v)
```
and
```python
ComponentUID('m.b.v')
```
will not be equal. I propose that CUIDs from slices follow the convention as those constructed from strings.

Reply:
I'm fine with this convention.